pkg/kubelet/sysctl/runtime.go

```diff
@@ -43,7 +44,7 @@ var _ lifecycle.PodAdmitHandler = &runtimeAdmitHandler{}
 
 // NewRuntimeAdmitHandler returns a sysctlRuntimeAdmitHandler which checks whether
 // the given runtime support sysctls.
-func NewRuntimeAdmitHandler(runtime container.Runtime) (*runtimeAdmitHandler, error) {
+func NewRuntimeAdmitHandler(runtime container.Runtime) (*sysctl.runtimeAdmitHandler, error) {
```

Comment:
As far as I can tell, nothing changed w.r.t. the definition of `runtimeAdmitHandler`. So I don't believe we need to change how we refer to the type.

Reply:
Thanks for your feedback!You are very paient to help me fix these problems. when reuturns type named *runtimeAdmitHandler,using golint tool occur some errors below:pkg/kubelet/sysctl/runtime.go:47:57: exported func NewRuntimeAdmitHandler returns unexported type *sysctl.runtimeAdmitHandler, which can be annoying to use.
I dont konw why Why golint treats this as warning?I am glad to hear your views.

Reply:
That's a great question ! The issue is relates to package visibility. Specifically, `NewRuntimeAdmitHandler` is an exported function, which means it can be called outside of the `sysctl` package. However, the `runtimeAdminHandler` type is unexported, so the type can only be referenced inside the `sysctl` package. The linter is remarking that its surprising that we have a public function return a private type.

In short, we can fix this error either by not exporting `NewRuntimeAdmitHandler`, exporting `runtimeAdmitHandler` directly, or making `NewRuntimeAdmitHandler` return a public interface which `runtimeAdmitHandler` fulfills.

If you're interested in learning more about this part of the code, please feel free to dig into the code and make a recommendation on a fix.

I'm also happy to share what I think the correct approach is (but I didn't want to take away a learning opportunity :) )

Reply:
hi,Matt!I need your help. it occurs little error,after  I followed your first suggestion . In short ,I changed  the case of the first letter of the identifier .
Specific log information:
pkg/kubelet/sysctl/runtime.go:43:5: cannot use &RuntimeAdmitHandler literal (type *RuntimeAdmitHandler) as type lifecycle.PodAdmitHandler in assignment:
	*RuntimeAdmitHandler does not implement lifecycle.PodAdmitHandler (missing Admit method)
I'd appreciate it if you could help me.

Reply:
Could you please push the most recent changes in a separate commit so that I can take a closer look? Thanks :)

Reply:
I have commit my lasted changes and I dont know if the correct or not. I am a new to approach this.please give me some advice .
I have another quesetion.look at pkg/kubelet/sysctl/runtime.go:38:6:.exported type RuntimeAdmitHandler have been comment. but the golint remain warning ,asking me comment on this var which is in line 40 .